.github/workflows/actions-compile.yml

```diff
@@ -18,16 +18,34 @@ jobs:
       - name: add missing packages
         uses: ConorMacBride/install-package@v1
         with:
-          apt: libelf-dev libbpf-dev libbsd-dev libsysrepo-dev libcap-dev libmnl-dev libselinux-dev libyang-tools
+          apt: libelf-dev libbpf-dev libbsd-dev libcap-dev libmnl-dev libselinux-dev cmake make gcc libtool-bin
+      - name: libyang v2.1.148 from upstream
+        uses: actions/checkout@v4
+        with:
+          repository: 'CESNET/libyang.git'
+          ref: 'v2.1.148'
+      - name: make libyang v2.1.148 from upstream
+        run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE:String="Release" .. && make -j $(nproc) && sudo make install
+      - name: sysrepo v2.2.150 from upstream
+        uses: actions/checkout@v4
+        with:
+          repository: 'sysrepo/sysrepo.git'
+          ref: 'v2.2.150'
+      - name: make sysrepo v2.2.150 from upstream
+        run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE:String="Release" .. && make -j $(nproc) && sudo make install
+      - name: run ldconfig
+        run: sudo ldconfig
+      - name: check sysrepo running ok
+        run: sudo ldconfig && sudo sysrepoctl -l
       - name: recursive checkout (with iproute2 submodule)
         uses: actions/checkout@v4
         with:
           submodules: true
       - name: make iproute2-sysrepo
-        run : CC=${{ matrix.compiler }} make -j $(nproc)
+        run:  CC=gcc make -j $(nproc)
       - name: basic compilation check
-        run : ldd bin/iproute2-sysrepo
+        run: ldd bin/iproute2-sysrepo
       - name: basic test ip address like
-        run : bin/iproute2-sysrepo link
+        run: bin/iproute2-sysrepo link
       - name: yanglint
-        run : make check
```

Comment:
Please, do not change the white spaces if not needed it unfocuses the code review.